stac_fastapi/testdata/joplin/hierarchy.json

```diff
@@ -0,0 +1,35 @@
+{
```

Comment:
Your scalability concern is reasonable. I'm having a hard time imagining it being a performance bottleneck, but JSON is brittle and it is (as currently imagined) directly maintained by developers (often a recipe for trouble). I'm not 100% sure what the best approach is to avoid such a difficulty. Perhaps rules for generating trees rather than fully specified trees? I'm open to suggestions.

Runtime addition of nodes should be possible with a small set of modifications. I agree that's important to add and am also in support of adding it in future updates